dolphin/db/sqlalchemy/api.py

```diff
@@ -303,25 +322,78 @@ def volume_get_all(context, marker=None, limit=None, sort_keys=None,
     return NotImplemented
 
 
-def pool_create(context, values):
+def pool_create(context, values, session=None):
     """Create a pool from the values dictionary."""
     return NotImplemented
 
 
-def pool_update(context, pool_id, values):
+def handle_db_data_error(f):
+    def wrapper(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except exception.InvalidRequest:
+            msg = _('Error writing field to database')
+            LOG.exception(msg)
+            raise exception.Invalid(msg)
+
+    return wrapper
+
+
+@handle_db_data_error
+def pool_update(context, pools):
     """Update a pool withe the values dictionary."""
-    return NotImplemented
+    pool_ref = models.Pool()
+    session = get_session()
+    with session.begin():
+        pool_refs = []
+        query = pool_get_all(context)
+        delete_list = copy.copy(query)
+        [delete_list.remove(item) for pool in pools
+         for item in query if pool.get('id') == item.get('id')]
+        for pool in delete_list:
+            session.delete(pool)
```

Comment:
problem is still here, the purpose of pool update is that update a pool, but the logic will update multiple pools and remove pools. plz check.